pkgs/development/python-modules/h5py/default.nix

```diff
@@ -39,10 +39,12 @@ buildPythonPackage rec {
     ./mpi4py-requirement.patch
   ];
 
-  # avoid strict pinning of numpy
+  # avoid strict pinning of numpy and mpi4py
   postPatch = ''
```

Comment:
I think this would be a good use of `pythonRelaxDepsHook`, see https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonrelaxdepshook-using-pythonrelaxdepshook

Reply:
> I think this would be a good use of `pythonRelaxDepsHook`, see [`master`/doc/languages-frameworks/python.section.md#using-pythonrelaxdepshook-using-pythonrelaxdepshook](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md?rgh-link-date=2024-07-17T12%3A15%3A58Z#using-pythonrelaxdepshook-using-pythonrelaxdepshook)

The peculiar thing about `pythonRelaxDepsHook` is that it doesn't work for dependencies defined in `pyproject.toml`, so for the numpy constraint it won't help. This has always bothered me so I opened https://github.com/NixOS/nixpkgs/issues/327941 and linked to it in the numpy comment.

Reply:
However, for the mpi4py dependency it did work.